### PR TITLE
C389473 (firebird)

### PR DESCRIPTION
**Pull request by PalinaDzmitrovich**
https://foliotest.testrail.io/index.php?/cases/view/389473

https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4375/allure/

### CODE REVIEW

**Review by ostapwd — APPROVED** [review]

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    );
+  });
+
+  after('Delete test data', () => {
+    Users.deleteViaApi(user.userId);
+  });
+
+  it(
+    'C389473 Verify that "Settings (Data export): display list of settings pages" permission is searchable and renamed (firebird) (Taas)',
+    { tags: [devTeams.firebird] },
```
Comment (dnovikau) [inline cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js:26]: TestTypes tag is missed

**Review by dnovikau — COMMENTED** [review]
Is it possible to replace cypress locators with interactors?

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js (hunk 1 above)
Comment (PalinaDzmitrovich) [reply cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js]: Fixed.

**Review by dnovikau — CHANGES_REQUESTED** [review]
Test is failing

File: cypress/support/fragments/data-export/data-export-permission.js
Comment (nayimovag) [inline cypress/support/fragments/data-export/data-export-permission.js:1]: Please remove this file. We have the necessary methods in fragments like `cypress\support\fragments\users\usersSearchPane.js`, `cypress\support\fragments\users\userEdit.js`, `cypress\support\fragments\users\usersCard.js` etc. You can look at implementation of C353978 and C350765 as an example

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import permissions from '../../support/dictionary/permissions';
+import TopMenu from '../../support/fragments/topMenu';
+import Users from '../../support/fragments/users/users';
+import devTeams from '../../support/dictionary/devTeams';
+import testTypes from '../../support/dictionary/testTypes';
+import dataExportPermission from '../../support/fragments/data-export/data-export-permission';
```
Comment (nayimovag) [inline cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js:6]: when importing fragments, please do so with upper case naming style, for example: DataExportPermission

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+let user;
+
+describe('Data-export', () => {
+  before('Create test data', () => {
+    cy.createTempUser([permissions.uiUsersView.gui, permissions.uiUsersPermissions.gui]).then(
+      (userProperties) => {
+        user = userProperties;
+        cy.login(user.username, user.password);
+        cy.visit(TopMenu.dataExportPath);
```
Comment (nayimovag) [inline cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js:16]: is it better to use login method this way? 
`
cy.login(user.username, user.password, {
          path: TopMenu.dataExportPath,
          waiter: TopMenu.dataExportPath,
        });
`

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import permissions from '../../support/dictionary/permissions';
+import TopMenu from '../../support/fragments/topMenu';
+import Users from '../../support/fragments/users/users';
+import devTeams from '../../support/dictionary/devTeams';
+import testTypes from '../../support/dictionary/testTypes';
+import dataExportPermission from '../../support/fragments/data-export/data-export-permission';
+
+let user;
+
+describe('Data-export', () => {
```
Comment (nayimovag) [inline cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js:10]: please change to 'data-export' so in allure report this test goes into the same folder with other data export tests

**Review by nayimovag — CHANGES_REQUESTED** [review]

File: cypress/support/fragments/data-export/data-export-permission.js
Comment (PalinaDzmitrovich) [reply cypress/support/fragments/data-export/data-export-permission.js:1]: Done.

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js (hunk 2 above)
Comment (PalinaDzmitrovich) [reply cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js]: Fixed.

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js (hunk 4 above)
Comment (PalinaDzmitrovich) [reply cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js]: Done.

File: cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js (hunk 3 above)
Comment (PalinaDzmitrovich) [reply cypress/e2e/data-export/display-list-of-settings-pages-permissions.cy.js]: Done.

**Comment by PalinaDzmitrovich** [comment]
> Test is failing

Fresh allure report:
https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4422/allure/

File: cypress/support/fragments/users/userEdit.js [hunk 5]
```diff
@@ -258,4 +265,49 @@ export default {
   selectSingleSelectValue: ({ data }) => {
     cy.do(Select({ label: data.fieldLabel }).choose(data.firstLabel));
   },
+
+  verifyUserPermissionsAccordion() {
+    cy.expect(permissionsAccordion.exists());
+    cy.expect(permissionsAccordion.has({ open: false }));
```
Comment (nayimovag) [inline cypress/support/fragments/users/userEdit.js:271]: if we have more than one assertion in a row, better to pass them as an array

File: cypress/support/fragments/users/userEdit.js [hunk 6]
```diff
@@ -258,4 +265,49 @@ export default {
...
+    permissionsList.perform((el) => {
+      el.invoke('attr', 'aria-rowcount').then((rowCount) => {
+        totalRows = rowCount;
+      });
+    });
+  },
+
+  openSelectPermissions() {
+    cy.do(permissionsAccordion.clickHeader());
+    cy.do(addPermissionsButton.click());
```
Comment (nayimovag) [inline cypress/support/fragments/users/userEdit.js:284]: if we have more than one command in a row, better to pass them as an array

File: cypress/support/fragments/users/userEdit.js [hunk 7]
```diff
@@ -258,4 +265,49 @@ export default {
...
+    cy.expect(selectPermissionsModal.exists());
+    this.permissionsCount();
+  },
+
+  searchForPermission(permission) {
+    cy.do(permissionsSearch.fillIn(permission));
+    cy.do(searchButton.click());
```
Comment (nayimovag) [inline cypress/support/fragments/users/userEdit.js:291]: if we have more than one command in a row, better to pass them as an array

**Review by nayimovag — APPROVED** [review]

File: cypress/support/fragments/users/userEdit.js (hunk 5 above)
Comment (PalinaDzmitrovich) [reply cypress/support/fragments/users/userEdit.js:271]: Only one assertion here, the accordion is present and collapsed by default

File: cypress/support/fragments/users/userEdit.js (hunk 6 above)
Comment (PalinaDzmitrovich) [reply cypress/support/fragments/users/userEdit.js:284]: This method opens a specific page

**Review by dnovikau — APPROVED** [review]

File: cypress/support/fragments/users/userEdit.js (hunk 5 above)
Comment (nayimovag) [reply cypress/support/fragments/users/userEdit.js:271]: i meant like 
`    cy.expect([
      permissionsAccordion.exists(),
      permissionsAccordion.has({ open: false })
    ]);`

**Pull request merged**